plugins/azure/batchAccounts/batchAccountsManagedIdentity.js

```diff
@@ -0,0 +1,52 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'Batch Account Managed Identity',
+    category: 'Batch',
+    domain: 'Compute',
+    severity: 'Medium',
+    description: 'Ensures that Batch accounts have managed identity enabled.',
+    more_info: 'Enabling managed identities eliminate the need for developers having to manage credentials by providing an identity for the Azure resource in Azure AD and using it to obtain Azure Active Directory (Azure AD) tokens.',
+    recommended_action: 'Modify Batch Account and enable managed identity.',
+    link: 'https://learn.microsoft.com/en-us/azure/batch/managed-identity-pools',
+    apis: ['batchAccounts:list'],
+    realtime_triggers: ['microsoftbatch:batchaccounts:write','microsoftbatch:batchaccounts:delete'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.batchAccounts, function(location, rcb){
+
+            var batchAccounts = helpers.addSource(cache, source,
+                ['batchAccounts', 'list', location]);
+
+            if (!batchAccounts) return rcb();
+
+            if (batchAccounts.err || !batchAccounts.data) {
+                helpers.addResult(results, 3, 'Unable to query for Batch accounts: ' + helpers.addError(batchAccounts), location);
+                return rcb();
+            }
+            if (!batchAccounts.data.length) {
+                helpers.addResult(results, 0, 'No existing Batch accounts found', location);
+                return rcb();
+            }
+
+            for (let batchAccount of batchAccounts.data) { 
+                if (!batchAccount.id) continue;
+        
+                if (batchAccount.identity && batchAccount.identity.type) {
+                    helpers.addResult(results, 0, 'Batch account has managed identity enabled', location, batchAccount.id);
```

Comment:
can we enable the managed identity as in whole on batch account or pools? from the link it seems like we can enable the identities for the pools in the batch account https://learn.microsoft.com/en-us/azure/batch/managed-identity-pools